client/src/api/block_builder/input_selection/utxo_chains/mod.rs

```diff
@@ -93,8 +96,28 @@ pub(crate) fn select_utxo_chain_inputs(
                     if !is_required_for_output && !allow_burning || is_required_for_input {
                         let nft_address = Address::Nft(NftAddress::new(nft_id));
                         let nft_required_in_unlock_condition = outputs.iter().any(|output| {
-                            // check if alias address is in unlock condition
-                            output_contains_address(output, None, &nft_address, current_time)
+                            if let Ok((required_unlock_address, unlocked_alias_or_nft_address)) = output
+                                .required_and_unlocked_address(
+                                    current_time,
+                                    // It's a new output, so the output id doesn't matter
+                                    OutputId::from_str(
```

Comment:
Use `OutputId::null` instead

Reply:
I tried that `no function or associated item named `null` found for struct `OutputId` in the current scope
function or associated item not found in `OutputId``

Reply:
Ah yeah it's there for all Ids but OutputId my bad